src/Microsoft.AspNet.Mvc.ModelBinding/Binders/KeyValuePairModelBinder.cs

```diff
@@ -40,7 +40,6 @@ public sealed class KeyValuePairModelBinder<TKey, TValue> : IModelBinder
             {
                 var untypedModel = propertyBindingContext.Model;
                 var model = ModelBindingHelper.CastOrDefault<TModel>(untypedModel);
-                parentBindingContext.ValidationNode.ChildNodes.Add(propertyBindingContext.ValidationNode);
```

Comment:
Forgot to mention, If a object gets created then yes both the properties will be validated ( which was the case before as well).